frame/support/src/genesis_builder_helper.rs

```diff
@@ -0,0 +1,49 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::traits::BuildGenesisConfig;
+use sp_genesis_builder::Result as BuildResult;
+use sp_runtime::format_runtime_string;
+
+/// Helper for implementing [`sp_genesis_builder::GenesisBuilder`] for runtimes.
+///
+/// Provides common logic. For more info refer to [`sp_genesis_builder::GenesisBuilder`].
+pub struct GenesisBuilderHelper<GC>(sp_std::marker::PhantomData<GC>);
```

Comment:
Actually not needed. Could be exposed as functions. Just wanted to encapsulate them into some entity (however mod is absolutely fine).

Reply:
I mean using a trait would maybe be more sensible?

Reply:
Exposing functions is fine, struct is not needed here for extra encapsulation.

It could be a trait but it would duplicate the functionality of the API already defined in [GenesisBuilder API](https://github.com/paritytech/substrate/blob/c32f5ed2ae6746d6f791f08cecbfc22fa188f5f9/primitives/genesis-builder/src/lib.rs#L38C2-L38C26). 

I can re-use the trait `runtime_decl_for_genesis_builder::GenesisBuilderV1<Block: sp_api::BlockT>` generated by macro. However `Block` is not really needed here for implementation.

Reply:
@bkchr 
Did you mean sth like this?
```
trait GenesisBuilderHelper<GC> {
  fn create_default_config() {...}
  fn build_config() {...}
}
impl<GC> GenesisBuilderHelper<GC> for () {}
```